src/models/highlightedtext.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include "highlightedtext.h"
+
+#include <QGuiApplication>
+#include <QPalette>
+
+#include <memory>
+
+#include <KColorScheme>
+#if KFSyntaxHighlighting_FOUND
+#include <KSyntaxHighlighting/AbstractHighlighter>
+#include <KSyntaxHighlighting/Definition>
+#include <KSyntaxHighlighting/Format>
+#include <KSyntaxHighlighting/Repository>
+#include <KSyntaxHighlighting/State>
+#include <KSyntaxHighlighting/SyntaxHighlighter>
+#include <KSyntaxHighlighting/Theme>
+#endif
+
+#include "formattingutils.h"
+
+namespace {
+#if KFSyntaxHighlighting_FOUND
+class KSyntaxHighlightingImplementation : public HighlightingImplementation,
```

Comment:
don't use the `K` prefix here, to ensure we don't run into name clashes in the future

Reply:
see comments in the header: I don't understand why you have this indirection at all. Just make this the `class HighlightingImplementation` and be done with it - no need for multiple inheritance or the `KSyntaxHighlightingImplementation` at all